packages/core/src/lib/upload.ts

```diff
@@ -15,18 +20,14 @@ export async function upload(
   options: UploadOptions,
   uploadFn: typeof uploadToPortal = uploadToPortal,
 ) {
-  if (options?.upload === undefined) {
-    throw new Error('upload config needs to be set');
-  }
-  const { apiKey, server, organization, project } = options.upload;
-  const { outputDir, filename } = options.persist;
+  const persist = normalizePersistConfig(options?.persist);
+  const upload = normalizeUploadConfig(options?.upload);
```

Comment:
As mentioned above, we can't supply defaults for the upload. The original implementation should stay in place here:

```suggestion
  if (!options?.upload) {
    throw new Error('upload config needs to be set');
  }
```